pkg/ext/stores/kubeconfig/store.go

```diff
@@ -0,0 +1,1487 @@
+func (s *Store) Create(
+	ctx context.Context,
+	obj runtime.Object,
+	createValidation rest.ValidateObjectFunc,
+	options *metav1.CreateOptions,
+) (runtime.Object, error) {
+	userInfo, _, isRancherUser, err := s.userFrom(ctx, "create")
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("error getting user info: %w", err))
+	}
+
+	if !isRancherUser {
+		return nil, apierrors.NewForbidden(gvr.GroupResource(), "", fmt.Errorf("user %s is not a Rancher user", userInfo.GetName()))
+	}
+
+	extras := userInfo.GetExtra()
+
+	authTokenID := first(extras[common.ExtraRequestTokenID])
+	if authTokenID == "" {
+		return nil, apierrors.NewForbidden(gvr.GroupResource(), "", fmt.Errorf("missing request token ID"))
+	}
+
+	authToken, err := s.tokenCache.Get(authTokenID)
+	if err != nil {
+		return nil, apierrors.NewForbidden(gvr.GroupResource(), "", fmt.Errorf("error getting request token %s: %w", authTokenID, err))
+	}
+
+	kubeconfig, ok := obj.(*ext.Kubeconfig)
+	if !ok {
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("invalid object type %T", obj))
+	}
+
+	if createValidation != nil {
+		if err := createValidation(ctx, obj); err != nil {
+			if _, ok := err.(apierrors.APIStatus); ok {
+				return nil, err
+			}
+			return nil, apierrors.NewBadRequest(fmt.Sprintf("create validation failed for kubeconfig: %s", err))
+		}
+	}
+
+	if !isUnique(kubeconfig.Spec.Clusters) {
+		return nil, apierrors.NewBadRequest("spec.clusters must be unique")
+	}
+
+	defaultTTL, err := s.getDefaultTTL()
+	if err != nil {
+		return nil, fmt.Errorf("error getting default token TTL: %w", err)
+	}
+	defaultTTLSeconds := *defaultTTL / 1000
+
+	ttlMilliseconds := kubeconfig.Spec.TTL * 1000
+	switch {
+	case ttlMilliseconds < 0:
+		return nil, apierrors.NewBadRequest("spec.ttl can't be negative")
+	case ttlMilliseconds == 0:
+		ttlMilliseconds = *defaultTTL
+		kubeconfig.Spec.TTL = defaultTTLSeconds
+	case ttlMilliseconds > *defaultTTL:
+		return nil, apierrors.NewBadRequest(fmt.Sprintf("spec.ttl %d exceeds max tll %d", kubeconfig.Spec.TTL, defaultTTLSeconds))
+	default: // Valid TTL.
+	}
+
+	host := s.getServerURL()
+	if host != "" {
+		u, err := url.Parse(host)
+		if err == nil {
+			host = u.Host
+		}
+	}
+	if host == "" {
+		host = first(extras[common.ExtraRequestHost])
+		if host == "" {
+			return nil, apierrors.NewInternalError(errors.New("can't determine the server URL"))
+		}
+	}
+
+	var (
+		conditions    []metav1.Condition
+		tokenIDs      []string
+		clusters      []*apiv3.Cluster
+		isAllClusters bool // User requested the kubeconfig for all clusters with "*".
+	)
+
+	localCluster, err := s.clusterCache.Get("local")
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("error getting local cluster: %w", err))
+	}
+
+	if len(kubeconfig.Spec.Clusters) > 0 {
+		if isAllClusters = kubeconfig.Spec.Clusters[0] == "*"; isAllClusters {
+			// The first id in the spec.clusters "*" means all clusters.
```

Comment:
Does this have to check that `len(kubeconfig.Spec.Clusters) == 1` as well ? Should it ?

Reply:
It can, but the original intent was exactly as expressed in this predicate - if the first element is `*` then get all clusters available to the user.